kubectl-plugin/src/resources/tests.rs

```diff
@@ -0,0 +1,178 @@
+use serde::ser;
+use std::{convert::TryInto, io::Read};
+use testlib::{Cluster, ClusterBuilder, Uuid};
+
+static CLUSTER: OnceCell<Cluster> = OnceCell::new();
+const VOLUME_UUID: &str = "1e3cf927-80c2-47a8-adf0-95c486bdd7b7";
+
+async fn setup() {
+    let cluster = ClusterBuilder::builder()
+        .with_rest(true)
+        .with_agents(vec!["core"])
+        .with_mayastors(1)
+        .with_pools(1)
+        .build()
+        .await
+        .unwrap();
+
+    CreateVolume {
+        uuid: VOLUME_UUID.try_into().unwrap(),
+        size: 5242880,
+        replicas: 1,
+        ..Default::default()
+    }
+    .request()
+    .await
+    .unwrap();
+
+    CLUSTER
+        .set(cluster)
+        .ok()
+        .expect("Expect to be initialised only once");
+}
+
+#[tokio::test]
+async fn get_volumes() {
+    if CLUSTER.get().is_none() {
+        setup().await;
+    }
+    let volumes = CLUSTER
+        .get()
+        .unwrap()
+        .rest_v00()
+        .volumes_api()
+        .get_volumes()
+        .await
+        .unwrap();
+    compare(format!(" ID                                    REPLICAS  TARGET-NODE  ACCESSIBILITY  STATUS  SIZE \n {}  1         <none>       <none>         Online  5242880 \n", VOLUME_UUID), volumes);
+}
+
+#[tokio::test]
+async fn get_volume() {
+    if CLUSTER.get().is_none() {
+        setup().await;
+    }
+    let volume = CLUSTER
+        .get()
+        .unwrap()
+        .rest_v00()
+        .volumes_api()
+        .get_volume(&Uuid::parse_str(VOLUME_UUID).unwrap())
+        .await
+        .unwrap();
+    compare(format!(" ID                                    REPLICAS  TARGET-NODE  ACCESSIBILITY  STATUS  SIZE \n {}  1         <none>       <none>         Online  5242880 \n", VOLUME_UUID), volume);
+}
+
+#[tokio::test]
+async fn get_pools() {
+    if CLUSTER.get().is_none() {
+        setup().await;
+    }
+    let pools = CLUSTER
+        .get()
+        .unwrap()
+        .rest_v00()
+        .pools_api()
+        .get_pools()
+        .await
+        .unwrap();
+    compare(format!(" ID                 TOTAL CAPACITY  USED CAPACITY  DISKS                                                                  NODE        STATUS  MANAGED \n mayastor-1-pool-1  100663296       8388608        {}  mayastor-1  Online  true \n", pools[0].state.as_ref().unwrap().disks[0].clone()), pools);
+}
+
+#[tokio::test]
+async fn get_pool() {
+    if CLUSTER.get().is_none() {
+        setup().await;
+    }
+    let pool = CLUSTER
+        .get()
+        .unwrap()
+        .rest_v00()
+        .pools_api()
+        .get_pool(CLUSTER.get().unwrap().pool(0, 0).as_str())
+        .await
+        .unwrap();
+    compare(format!(" ID                 TOTAL CAPACITY  USED CAPACITY  DISKS                                                                  NODE        STATUS  MANAGED \n mayastor-1-pool-1  100663296       8388608        {}  mayastor-1  Online  true \n", pool.state.as_ref().unwrap().disks[0].clone()), pool);
```

Comment:
Wherever possible we should retrieve the values from objects, eg:
mayastor-1-pool-1 can be set to {} and retrieved from pool.id 
mayastor-1  can be set to {} and retrieved from pool.state.node
Online  can be set to {} and retrieved from pool.state.status 
or something like that

Reply:
Made the change. Thanks